node/bft-consensus/src/lib.rs

```diff
@@ -186,3 +192,37 @@ fn read_authority_keypair_from_file<P: AsRef<std::path::Path>>(path: P) -> anyho
     let contents = std::fs::read_to_string(path)?;
     BLS12381KeyPair::decode_base64(contents.as_str().trim()).map_err(|e| anyhow!(e))
 }
+
+#[derive(Clone)]
+struct TransactionValidator<N: Network, C: ConsensusStorage<N>>(AleoConsensus<N, C>);
+
+impl<N: Network, C: ConsensusStorage<N>> narwhal_worker::TransactionValidator for TransactionValidator<N, C> {
+    type Error = anyhow::Error;
+
+    /// Determines if a transaction valid for the worker to consider putting in a batch
+    fn validate(&self, transaction: &[u8]) -> Result<(), Self::Error> {
```

Comment:
This implementation could use some tests, at least for a happy path and a fail case.

Reply:
in order to such a test to be most accurate, I think it would need to be a full integration test :thinking:; those will follow as soon as possible

Reply:
>  those will follow as soon as possible

In this PR or separate one? If the latter, happy to +1 this PR :slightly_smiling_face:

Reply:
in a latter one, as those might take a while longer to create